### PR TITLE
Add missing history for invoice + bug fixes

### PR DESCRIPTION
**Pull request by sbrossie**

### CODE REVIEW

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java [hunk 1]
```diff
@@ -143,6 +150,21 @@ public boolean apply(final InvoicePayment input) {
         return Response.status(Response.Status.OK).entity(result).build();
     }
 
+    @TimedResource
+    @GET
+    @Path("/{invoicePaymentId:" + UUID_PATTERN + "}/" + AUDIT_LOG_WITH_HISTORY)
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Retrieve invoice payment audit logs with history by id", response = AuditLogJson.class, responseContainer = "List")
+    @ApiResponses(value = {@ApiResponse(code = 404, message = "Invoice payment not found")})
+    public Response getInvoiceItemAuditLogsWithHistory(@PathParam("invoicePaymentId") final UUID invoicePaymentId,
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java]: Naming to update.

File: util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java [hunk 2]
```diff
@@ -373,8 +373,7 @@ private static String getKey(final String rawKey, final CacheType cacheType, fin
                     // Make sure to re-hydrate the objects first (especially needed for create calls)
                     final Collection<M> reHydratedEntities = new ArrayList<>(entityRecordIds.size());
                     if (deletedAndUpdatedEntities.isEmpty()) {
-                        Preconditions.checkState(entityRecordIds.size() == 1 && changeType == ChangeType.INSERT, "Unexpected number of entityRecordIds=%s and changeType=%s", entityRecordIds, changeType);
-                        reHydratedEntities.add(sqlDao.getByRecordId(entityRecordIds.get(0), context));
+                        reHydratedEntities.addAll((List<M>) sqlDao.getByRecordIds(entityRecordIds, context));
```
Comment (pierre) [inline util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java]: Out of curiosity -- why this change (just curious why the `Preconditions` has changed)? Is that for bulk insert?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java (hunk 1 above)
Comment (sbrossie) [reply jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java]: Ack, will fix -- and regen + re-release client

File: util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java (hunk 2 above)
Comment (sbrossie) [reply util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java]: As discussed, bulk invoice item creation.
